docs/404.rst

```diff
@@ -0,0 +1,14 @@
+.. _404:
```

Comment:
maybe also a mention of if the user was switching from <= 2.5 to 2.6, the paths changed so they will want to go to the [main page](link) and navigate to the page they want?